### PR TITLE
Adding support for storing and rendering keypoints and polylines

### PR DESCRIPTION
**Pull request by brimoor**
Addresses https://github.com/voxel51/fiftyone/issues/560. Keypoints and polylines can now be stored on samples, and they can be rendered on images via `SampleCollection.draw_labels()` and visualized in the App.

Example usage:

```py
import random

import fiftyone as fo

IMAGE_PATH = "/path/to/white-image.png"
ANNO_DIR = "/path/for/annos"

sample = fo.Sample(filepath=IMAGE_PATH)

# Add keypoint
sample["keypoint"] = fo.Keypoint(
    label="single",
    points=[(random.random(), random.random()) for _ in range(10)],
    attributes={
        "number": fo.NumericAttribute(value=51)
    },
)

# Add keypoints
sample["keypoints"] = fo.Keypoints(
    keypoints=[
        fo.Keypoint(
            label="random",
            points=[(random.random(), random.random()) for _ in range(10)],
        )
    ]
)

# Polyline
sample["polyline"] = fo.Polyline(
    label="square",
    points=[(0.1, 0.1), (0.3, 0.1), (0.3, 0.3), (0.1, 0.3)],
    closed=True,
    filled=True,
    attributes={
        "name": fo.CategoricalAttribute(value="FiftyOne")
    },
)

# Add polylines
sample["polylines"] = fo.Polylines(
    polylines=[
        fo.Polyline(
            points=[(0.3, 0.3), (0.7, 0.3), (0.7, 0.7)],
            closed=False,
            filled=False,
        ),
        fo.Polyline(
            label="triangle",
            points=[(0.4, 0.4), (0.6, 0.4), (0.6, 0.6)],
            closed=True,
            filled=True,
        )
    ]
)

dataset = fo.Dataset()
dataset.add_sample(sample)

print(sample)

annotation_config = foua.AnnotationConfig(
    {
        "show_polyline_names": False,
        "show_keypoints_names": False,
        "per_polyline_label_colors": True,
        "per_keypoints_label_colors": True,
    }
)

dataset.draw_labels(ANNO_DIR, annotation_config=annotation_config)
```

Example annotated image:

test-white

One benefit of this PR is that we can now load the polyline labels of the BDD format, which were previously ignored. This PR adds support for that too.

```py
import fiftyone as fo
import fiftyone.utils.annotations as foua

DATASET_DIR = "/path/to/bdd100k/validation"
ANNO_DIR = "/path/for/annos"

dataset = fo.Dataset.from_dir(
    DATASET_DIR,
    fo.types.BDDDataset,
    expand=True,
    shuffle=True,
    max_samples=25,
)

dataset.draw_labels(ANNO_DIR, overwrite=True)
```

Here's an example of a BDD image loaded + rendered in all its glory:

bf02cf02-6168f0d6

And here's a custom rendering that's more readable:

```py
# More readable output
annotation_config = foua.AnnotationConfig({
    "per_object_label_colors": True,
    "per_polyline_label_colors": True,
    "hide_non_filled_polyline_annos": True,
    "show_object_names": False,
    "show_object_attrs": False,
    "show_polyline_names": False,
    "show_polyline_attrs": False,
})

dataset.draw_labels(
    ANNO_DIR, overwrite=True, annotation_config=annotation_config
)
```

b8781f69-81bcdb6a

### CODE REVIEW

**Comment by lethosor** [comment]
Looks like the dependency on https://github.com/voxel51/eta/pull/480 is causing test failures (possibly all of them). There isn't currently a way to run the tests on GitHub with a specific ETA build, but since those changes are strictly additive (as far as I can tell), it should be fine to merge the ETA PR and put up a new ETA build for everyone, then re-run tests here.

Before we try to do that, is this PR targeted for 0.6.0? I'm fine with merging the ETA PR as soon as feasible, as long as it maintains backwards compatibility, but frontend support for these structures won't be ready for 0.6.0, so it may not be worth the effort at this point.

**Comment by brimoor** [comment]
This PR is ready for frontend support as well. I may also add support for loading polylines from COCO and CVAT formats in this PR, since they can contain those as well.

The tests pass locally with the correct ETA version.

Not planning on including this in v0.6.0 either. We can save for the following release. There shouldn't be anything breaking in the ETA PR, but lets hold that until after v0.6.0 too.

**Comment by lethosor** [comment]
The couple merges of #577 into this PR seem to have broken tests - does this PR depend on CVAT video?

File: fiftyone/core/labels.py [hunk 1]
```diff
@@ -129,18 +201,10 @@ class Classification(ImageLabel):
 
     meta = {"allow_inheritance": True}
 
-    _id = fof.ObjectIdField(
-        required=True, default=ObjectId, unique=True, primary_key=True
-    )
```
Comment (lethosor) [inline fiftyone/core/labels.py:134]: I'm able to reliably reproduce a failure in `SerializationTests.test_embedded_document` locally (with `pytest tests/unittests.py -k test_embedded_document`).
I'm guessing the issue is that you've removed the `_id` field here but haven't made `Classification` inherit from `_HasID`. Not sure what else might need to change here.

Comment (brimoor) [reply fiftyone/core/labels.py:134]: Thanks for catching this! I fixed it

**Comment by lethosor** [comment]
FYI: I'm working on resolving several merge conflicts introduced by the merge of #555

**Comment by lethosor** [comment]
Frontend support here should be good to go

**Review by benjaminpkane — APPROVED** [review]
Can confirm keypoints and polylines are both working in the App! Cool stuff!

**Review by lethosor — CHANGES_REQUESTED** [review]
Somehow the docs build started failing between https://github.com/voxel51/fiftyone/pull/562/commits/b19e2373ec5b7e8e3e3863256d8ed7f433c12df6 and https://github.com/voxel51/fiftyone/pull/562/commits/e9957da1cf10de24961ac624d6843cef6530ec4f. Not seeing anything obvious, but it only occurs on this branch: https://github.com/voxel51/fiftyone/actions?query=workflow%3A%22Build+docs%22+is%3Afailure

I'm investigating, but we shouldn't merge this yet.

**Review by lethosor — APPROVED** [review]

**Pull request merged**